plugins/woocommerce/src/Admin/API/Reports/Orders/DataStore.php

```diff
@@ -32,6 +31,13 @@ public function __construct() {
 		parent::__construct();
 	}
 
+	/**
+	 * Set up all the hooks for maintaining data consistency (transients and co).
+	 */
+	public static function init() {
+		add_action( 'woocommerce_analytics_update_order_stats', array( __CLASS__, 'actualize_all_statuses_transient_if_necessary' ) );
```

Comment:
Nice!  This is a creative way to keep the cache up to date.

Reply:
Thanks!